src/main/java/backend/resource/TurboLabel.java

```diff
@@ -197,18 +286,65 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         TurboLabel that = (TurboLabel) o;
-        return actualName.equals(that.actualName) && colour.equals(that.colour);
+        return fullName.equals(that.fullName) && colour.equals(that.colour);
     }
 
     @Override
     public int hashCode() {
-        int result = actualName.hashCode();
+        int result = fullName.hashCode();
         result = 31 * result + colour.hashCode();
         return result;
     }
 
     @Override
     public int compareTo(TurboLabel o) {
-        return actualName.compareTo(o.getActualName());
+        return fullName.compareTo(o.getFullName());
+    }
+
+    private static String buildFullName(String group, String name, boolean isExclusive) {
+        return group + (isExclusive ? EXCLUSIVE_DELIMITER : NONEXCLUSIVE_DELIMITER) + name;
+    }
+
+    private Grouping determineGrouping() {
+        if (!getDelimiter(fullName).isPresent()) return Grouping.NONE;
+
+        return getDelimiter(fullName).get().equals(EXCLUSIVE_DELIMITER) 
+                ? Grouping.EXCLUSIVE : Grouping.NON_EXCLUSIVE;
+    }
+
+    /**
+     * Extracts shortName and groupName from a keyword
+     * groupName is name of the group associated with a keyword
+     * shortName is name of the keyword after omitting its group name 
+     * Example: "priority.high" -> groupName = "priority", shortName = "high"
+     * @param keyword
+     * @return shortName and groupName of a given keyword
+     */
+    public static String[] extractNames(String keyword) {
+        if (!getDelimiter(keyword).isPresent()) return new String[] {"", keyword};
+
+        int delimiterIndex = keyword.indexOf(getDelimiter(keyword).get());
+        String shortName = keyword.substring(delimiterIndex + 1);
+        String groupName = keyword.substring(0, delimiterIndex);
```

Comment:
All of the examples should not pass the check. I don't think we should throw any exception for parsing of `shortName` and `fullName` since grouping is `optional`.